lib/_stream_wrap.js

```diff
@@ -10,9 +10,11 @@ function StreamWrap(stream) {
 
   this.stream = stream;
 
+  this.queue = null;
```

Comment:
Totally a noob question. Why are we not implementing this with an array instead of this complex enqueue and dequeue logic?